src/AWS/Encode.elm

```diff
@@ -0,0 +1,34 @@
+module AWS.Encode exposing (..)
+
+import Char
+import Hex
+import Regex exposing (regex, HowMany(All))
+
+
+{-| We don't use Http.encodeUri because it misses some characters.
+
+For example, ! ' ( ) etc.
+
+Only "Unreserved Characters" are allowed.
+See http://tools.ietf.org/html/rfc3986
+Section 2.3
+-}
+uri : String -> String
```

Comment:
Extracted `Canonical.encodeUri` into its own module so that `Http` could also using it, without causing a dependency cycle.